### ISSUE TITLE
watchQuery defaultOptions are unset when query variables change (specifically nextFetchPolicy)

### ISSUE THREAD

**Issue by jebonfig** [apollographql/apollo-client#6839]
**Intended outcome:**

The `defaultOptions.watchQuery.nextFetchPolicy` setting should always be applied for the query, even if the query variables change.  I am setting it to `cache-only` to retain the behavior from https://github.com/apollographql/apollo-client/pull/6353

**Actual outcome:**

When the query variables change, the `defaultOptions` are unset.  This results in unexpected network calls after cached data changes.

**How to reproduce the issue:**

Simple repro here: https://github.com/jebonfig/react-apollo-error-template

**Versions**

```
  System:
    OS: macOS High Sierra 10.13.6
  Binaries:
    Node: 11.2.0 - /usr/local/bin/node
    npm: 6.4.1 - /usr/local/bin/npm
  Browsers:
    Chrome: 84.0.4147.125
    Safari: 13.1.2
  npmPackages:
    @apollo/client: ^3.1.3 => 3.1.3 
```

**Comment by benjamn** [comment]
This does seem important, and not just for `nextFetchPolicy` (though see https://github.com/apollographql/apollo-client/issues/6833#issuecomment-674971155 for some other ideas about that).

**Comment by dylanwulf** [comment]
@benjamn Hi, I hate to be the person to post a "+1 me too", but this issue has been out here for 8 months now and is still a problem. It looks like @jebonfig has updated their reproduction to use the new `nextFetchPolicy` function, but it still gets unset whenever the variables change. 
Also just want to say thank you for all your hard work; I've seen how many issues get posted on this repo every day and it must be difficult keeping up with all of them.

**Comment by benjamn** [comment]
@jebonfig Please try `@apollo/client@3.4.0-rc.18` when you have a chance. Sorry for the wait!

**Comment by jebonfig** [comment]
Thanks @benjamn!  This is fixed on `3.4.0`.
